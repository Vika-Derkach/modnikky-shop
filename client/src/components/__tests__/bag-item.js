//import "jest-dom/extend-expect";
import "@testing-library/jest-dom/extend-expect";
import { cleanup, render } from "@testing-library/react";
import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import BagItem from "../bag-item";

afterEach(() => {
  cleanup();
});

it("renders without crashing", () => {
  const product = {
    frontPicture: "https://i.ytimg.com/vi/1Ne1hqOXKKI/maxresdefault.jpg",
    title: "cat",
    price: 43,
    color: "GREY",
    count: 1,
    size: "L",
  };
  const div = document.createElement("div");
  ReactDOM.render(<BagItem product={product} />, div);
});

it("renders button correctly", () => {
  const product = {
    frontPicture:
      "https://assets.burberry.com/is/image/Burberryltd/EDC85896-F540-490D-B9A4-5ADF6A0EED21.jpg?$BBY_V2_ML_1x1$&wid=2500&hei=2500",
    title: "Coat",
    price: 43,
    color: "GREY",
    count: 1,
    size: "L",
  };
  const { getByTestId } = render(<BagItem product={product} />);
  const newBagItem = getByTestId("bag-item");

  expect(newBagItem).toBeInTheDocument();
  expect(newBagItem).toHaveTextContent("Coat");
  expect(newBagItem).not.toContainHTML("strike");
  expect(newBagItem).toContainHTML("img");
});
it("matches snapshot 1", () => {
  const product = {
    frontPicture:
      "https://assets.burberry.com/is/image/Burberryltd/EDC85896-F540-490D-B9A4-5ADF6A0EED21.jpg?$BBY_V2_ML_1x1$&wid=2500&hei=2500",
    title: "Coat",
    price: 43,
    color: "GREY",
    count: 1,
    size: "L",
  };
  const tree = renderer.create(<BagItem product={product} />).toJSON();
  expect(tree).toMatchSnapshot();
});
it("matches snapshot 2", () => {
  const product = {
    frontPicture:
      "https://www.politix.com.au/dw/image/v2/ABBA_PRD/on/demandware.static/-/Sites-politix-master-catalog/default/dw68075a91/images/hires/Winter%2021/A3/ac07-black-1.jpg?sw=2000&sh=2000&sm=fit",
    title: "Coat",
    price: 43,
    color: "Blue",
    count: 1,
    size: "M",
  };
  const tree = renderer.create(<BagItem product={product} />).toJSON();
  expect(tree).toMatchSnapshot();
});
