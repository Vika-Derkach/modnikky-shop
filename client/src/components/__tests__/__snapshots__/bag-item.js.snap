// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<div
  className="bag-item"
  data-testid="bag-item"
>
  <div
    className="bag-item_picture"
  >
    <img
      alt=""
      src="https://assets.burberry.com/is/image/Burberryltd/EDC85896-F540-490D-B9A4-5ADF6A0EED21.jpg?$BBY_V2_ML_1x1$&wid=2500&hei=2500"
    />
  </div>
  <div
    className="bag-item_descr"
  >
    <div
      className="bag-item_descr__title"
    >
      Coat
    </div>
    <div
      className="bag-item_descr__price"
    >
      USD $
      43
    </div>
    <div
      className="bag-item_descr__selector"
    >
      COLOR: 
      GREY
    </div>
    <div
      className="bag-item_descr__selector"
    >
      SIZE: 
      L
       
    </div>
     
    <div
      className="bag-item_descr__selector"
    >
      QUANTITY:
       
      <button
        className="button-update button-quantity"
      >
        <i
          aria-hidden="true"
          className="fa fa-minus"
        />
      </button>
      <span
        className="quantity-number"
      >
        1
      </span>
      <button
        className="button-update button-quantity"
      >
        <i
          aria-hidden="true"
          className="fa fa-plus"
        />
      </button>
    </div>
  </div>
  <button
    className="button-update"
  >
    <i
      aria-hidden="true"
      className="fa fa-times"
    />
  </button>
</div>
`;

exports[`matches snapshot 2 1`] = `
<div
  className="bag-item"
  data-testid="bag-item"
>
  <div
    className="bag-item_picture"
  >
    <img
      alt=""
      src="https://www.politix.com.au/dw/image/v2/ABBA_PRD/on/demandware.static/-/Sites-politix-master-catalog/default/dw68075a91/images/hires/Winter%2021/A3/ac07-black-1.jpg?sw=2000&sh=2000&sm=fit"
    />
  </div>
  <div
    className="bag-item_descr"
  >
    <div
      className="bag-item_descr__title"
    >
      Coat
    </div>
    <div
      className="bag-item_descr__price"
    >
      USD $
      43
    </div>
    <div
      className="bag-item_descr__selector"
    >
      COLOR: 
      Blue
    </div>
    <div
      className="bag-item_descr__selector"
    >
      SIZE: 
      M
       
    </div>
     
    <div
      className="bag-item_descr__selector"
    >
      QUANTITY:
       
      <button
        className="button-update button-quantity"
      >
        <i
          aria-hidden="true"
          className="fa fa-minus"
        />
      </button>
      <span
        className="quantity-number"
      >
        1
      </span>
      <button
        className="button-update button-quantity"
      >
        <i
          aria-hidden="true"
          className="fa fa-plus"
        />
      </button>
    </div>
  </div>
  <button
    className="button-update"
  >
    <i
      aria-hidden="true"
      className="fa fa-times"
    />
  </button>
</div>
`;
